### PR TITLE
Machine end

### PR DESCRIPTION
**Pull request by EliosPride**

### CODE REVIEW

File: src/main/java/javaoop/Main.java [hunk 1]
```diff
@@ -1,4 +1,15 @@
 package javaoop;
 
 public class Main {
-}
+    public static void main(String[] args) {
+        Truck truck = new Truck();
+        Bulldozer bulldozer = new Bulldozer();
+        Excavator excavator = new Excavator();
```
Comment (al-pavlenko) [inline src/main/java/javaoop/Main.java:7]: @EliosPride your machines must be of Machine type

File: src/main/java/javaoop/Main.java [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+        truck.doWork();
+        bulldozer.doWork();
+        excavator.doWork();
```
Comment (al-pavlenko) [inline src/main/java/javaoop/Main.java:10]: @EliosPride collect all machines to array and run each method in the loop

File: src/main/java/javaoop/Excavator.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package javaoop;
+
+public class Excavator extends Machine{
```
Comment (al-pavlenko) [inline src/main/java/javaoop/Excavator.java:3]: @EliosPride use ctrl+alt+L to refactor code style

File: src/main/java/javaoop/Bulldozer.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package javaoop;
+
+public class Bulldozer extends Machine{
```
Comment (al-pavlenko) [inline src/main/java/javaoop/Bulldozer.java:3]: @EliosPride refactor code style

File: src/main/java/javaoop/Main.java [hunk 5]
```diff
@@ -1,4 +1,15 @@
...
+        truck.stopWork();
+        bulldozer.stopWork();
+        excavator.stopWork();
+    }
+}
```
[the remark this answers is not in the document]
Comment (EliosPride) [reply src/main/java/javaoop/Main.java]: @al-pavlenko done, gone done

File: src/main/java/javaoop/Main.java (hunk 1 above)
Comment (EliosPride) [reply src/main/java/javaoop/Main.java]: @al-pavlenko done, gone done

File: src/main/java/javaoop/Main.java (hunk 2 above)
Comment (EliosPride) [reply src/main/java/javaoop/Main.java]: @al-pavlenko done, gone done

File: src/main/java/javaoop/Excavator.java (hunk 3 above)
Comment (EliosPride) [reply src/main/java/javaoop/Excavator.java]: @al-pavlenko done, gone done

File: src/main/java/javaoop/Bulldozer.java (hunk 4 above)
Comment (EliosPride) [reply src/main/java/javaoop/Bulldozer.java]: @al-pavlenko done, gone done

**Comment by EliosPride** [comment]
@al-pavlenko done, gone done

File: src/main/java/javaoop/Main.java [hunk 6]
```diff
@@ -2,14 +2,13 @@
...
         Excavator excavator = new Excavator();
-        truck.doWork();
-        bulldozer.doWork();
-        excavator.doWork();
-        truck.stopWork();
-        bulldozer.stopWork();
-        excavator.stopWork();
+        Machine[] array = {truck, bulldozer, excavator};
+        for (Machine work : array) {
+            work.doWork();
```
Comment (al-pavlenko) [inline src/main/java/javaoop/Main.java:10]: @EliosPride doWork() and stopWork() must be in separate loops

File: src/main/java/javaoop/Main.java [hunk 7]
```diff
@@ -2,14 +2,13 @@
 
 public class Main {
     public static void main(String[] args) {
-        Truck truck = new Truck();
+        Machine truck = new Truck();
```
Comment (al-pavlenko) [inline src/main/java/javaoop/Main.java:5]: @EliosPride ok, you changed to Machine object Truck, but what about other machines?

**Pull request closed without merging**